File: admin/UV Charging/src/models/NotificationModel.js
import mongoose from "mongoose";

const NotificationSchema = new mongoose.Schema(
  {
    userId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true,
      unique: true,
    },
    chargingStatusAlerts: { type: Boolean, default: false },
    lowBatteryAlerts: { type: Boolean, default: false },
    bookingUpdates: { type: Boolean, default: false },
    stationUpdates: { type: Boolean, default: false },
    paymentAndSession: { type: Boolean, default: false },
  },
  {
    timestamps: true,
    versionKey: false,
  }
);

const NotificationModel = mongoose.model("Notification", NotificationSchema);
export default NotificationModel;

File: mobile/lib/feature/Notification/model/notification_model.dart
class NotificationModel {
  String? status;
  Data? data;

  NotificationModel({this.status, this.data});

  NotificationModel.fromJson(Map<String, dynamic> json) {
    status = json['status'];
    data = json['data'] != null ? new Data.fromJson(json['data']) : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['status'] = this.status;
    if (this.data != null) {
      data['data'] = this.data!.toJson();
    }
    return data;
  }
}

class Data {
  bool? chargingStatusAlerts;
  bool? lowBatteryAlerts;
  bool? bookingUpdates;
  bool? stationUpdates;
  bool? paymentAndSession;

  Data(
      {this.chargingStatusAlerts,
        this.lowBatteryAlerts,
        this.bookingUpdates,
        this.stationUpdates,
        this.paymentAndSession});

  Data.fromJson(Map<String, dynamic> json) {
    chargingStatusAlerts = json['chargingStatusAlerts'];
    lowBatteryAlerts = json['lowBatteryAlerts'];
    bookingUpdates = json['bookingUpdates'];
    stationUpdates = json['stationUpdates'];
    paymentAndSession = json['paymentAndSession'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['chargingStatusAlerts'] = this.chargingStatusAlerts;
    data['lowBatteryAlerts'] = this.lowBatteryAlerts;
    data['bookingUpdates'] = this.bookingUpdates;
    data['stationUpdates'] = this.stationUpdates;
    data['paymentAndSession'] = this.paymentAndSession;
    return data;
  }
}
